packages/react/src/components/NumberInput/NumberInput.js

```diff
@@ -194,7 +199,14 @@ class NumberInput extends Component {
     if (!disabled) {
       evt.persist();
       evt.imaginaryTarget = this._inputRef;
-      const value = evt.target.value;
+      let value = evt.target.value;
+      if (enforceValidation) {
+        if (value > max) {
+          value = max;
+        } else if (value < min) {
+          value = min;
```

Comment:
You need to add these to the prop extract statement ie
`const { disabled, onChange, min, max, enforceValidation } = this.props;`